elastic/apm/tasks/setup.json

```diff
@@ -0,0 +1,47 @@
+{
+  "name": "delete-all-data-streams",
+  "operation": {
+      "operation-type": "delete-data-stream",
+      "data-stream": [
+        "traces-apm*",
+        "metrics-apm*",
+        "logs-apm*"
+      ]
+  }
+},
+{
+  "name": "delete-all-composable-templates",
+  "operation": {
+    "operation-type": "delete-composable-template"
+  }
+},
+{
+  "name": "delete-all-component-templates",
+  "operation": {
+    "operation-type": "delete-component-template"
+  }
+},
+{        
+  "name": "create-all-component-templates",
+  "operation": {
+    "operation-type": "create-component-template"
+  }
+},
```

Comment:
Don't these operations delete all the default templates for apm? Or is this called before the ES apm plugin is loaded? I didn't look at any order.

Reply:
`delete-*` operations will delete the templates specified in the `track.json` specification. So yes if we are going to use the same name for a component(e.g. `apm@settings` as suggested in the other comment) we shouldn't delete the component templates but just update them. Thanks for pointing this out, I will address this  🙏